dns/dns.go

```diff
@@ -21,41 +24,68 @@ const (
 
 // Config provides the necessary configuration for creating a Collector.
 type Config struct {
-	Logger micrologger.Logger
+	KubernetesClient kubernetes.Interface
```

Comment:
```suggestion
	K8sClient kubernetes.Interface
```

Reply:
Will take care in follow up, this is also differently named in the network collector, see https://github.com/giantswarm/giantswarm/issues/7823 for tracking